staging/src/k8s.io/apiserver/pkg/util/flag/map_string_string.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// MapStringString is a shim for setting map[string]string values from the command line
+type MapStringString map[string]string
+
+// String implements github.com/spf13/pflag.Value
+func (m MapStringString) String() string {
+	pairs := []string{}
+	for k, v := range m {
+		pairs = append(pairs, fmt.Sprintf("%s=%s", k, v))
+	}
+	sort.Strings(pairs)
+	return strings.Join(pairs, ",")
+}
+
+// Set implements github.com/spf13/pflag.Value
+func (m MapStringString) Set(value string) error {
```

Comment:
The old eviction logic had checking for duplication.  How do we want to handle that in the kubelet config?

Reply:
You can't have duplicate keys in a map. 

I guess you could set multiple keys via CLI, and end up having later keys override earlier keys. I'm not convinced that this is a big deal.